cpp/src/Ice/SSL/DistinguishedName.h

```diff
@@ -2,78 +2,17 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-#ifndef ICE_SSL_CERTIFICATE_H
-#define ICE_SSL_CERTIFICATE_H
+#ifndef ICE_SSL_DISTINGUISHED_NAME_H
+#define ICE_SSL_DISTINGUISHED_NAME_H
 
-#include "../Config.h"
-#include "../Exception.h"
-#include "ConnectionInfoF.h"
+#include "Ice/Config.h"
 
-#include <chrono>
-#include <cstdint>
-#include <functional>
 #include <list>
-#include <vector>
+#include <string>
+#include <utility>
 
 namespace Ice::SSL
 {
-    /**
-     * Thrown if the certificate cannot be read.
-     */
-    class ICE_API CertificateReadException : public Ice::Exception
-    {
-    public:
-        using Ice::Exception::Exception;
-
-        CertificateReadException(const char*, int, std::string) noexcept;
-
-        std::string ice_id() const override;
-
-        /** The reason for the exception. */
-        std::string reason;
-
-    private:
-        static const char* _name;
-    };
-
-    /**
-     * Thrown if the certificate cannot be encoded.
-     */
-    class ICE_API CertificateEncodingException : public Ice::Exception
-    {
-    public:
-        using Ice::Exception::Exception;
-
-        CertificateEncodingException(const char*, int, std::string) noexcept;
-
-        std::string ice_id() const override;
-
-        /** The reason for the exception. */
-        std::string reason;
-
-    private:
-        static const char* _name;
-    };
-
-    /**
-     * This exception is thrown if a distinguished name cannot be parsed.
-     */
-    class ICE_API ParseException : public Ice::Exception
-    {
-    public:
-        using Ice::Exception::Exception;
-
-        ParseException(const char*, int, std::string) noexcept;
-
-        std::string ice_id() const override;
-
-        /** The reason for the exception. */
-        std::string reason;
-
-    private:
-        static const char* _name;
-    };
-
     /**
```

Comment:
Since this class doesn't appear to be used anywhere outside the Ice code, you should remove all the ICE_API.

Reply:
You are right, don't need to export this class.

Reply:
Actually this is used in IceGrid implementation

https://github.com/zeroc-ice/ice/blob/dc87569b5e0e462fb7a00830d59185ead7804c84/cpp/src/IceGrid/InternalRegistryI.cpp#L65-L106

Seems we will be able to remove when we fix https://github.com/zeroc-ice/ice/issues/2135